lib/iris/fileformats/netcdf/saver.py

```diff
@@ -2509,6 +2583,39 @@ def store(data, cf_var, fill_value):
             )
             warnings.warn(msg.format(cf_var.name, fill_value))
 
+    def _deferred_save(self):
+        """
+        Create a 'delayed' to trigger file completion for lazy saves.
+
+        This contains all the deferred writes, which complete the file by filling out
+        the data of variables initially created empty.
+
+        """
+        # Create a lock to satisfy the da.store call.
+        # We need a serialisable lock for scheduling with processes or distributed.
+        # See : https://github.com/dask/distributed/issues/780
+        # However, this does *not* imply safe access for file writing in parallel.
+        # For that, DeferredSaveWrapper uses a filelock as well.
+        lock = SerializableLock()
```

Comment:
As you mention, the synchronisation of the parallel writing should be taken care of by the DeferredSaveWrapper, so one could set `lock=False` here.

Reply:
Thanks for the hint.  The truth is, we are (I am) still learning here.
And also, it looks like the #2063 issue will affect the eventual implementation of this -- this code is definitely only a draft "Proof of Concept" solution.

Reply:
Sure! Just thought I would write this down, since it also took me a while to figure out from Xarray..